src/main/java/ua/lviv/iot/parkingServer/service/ParkingService.kt

```diff
@@ -1,51 +1,28 @@
 package ua.lviv.iot.parkingServer.service
 
-import jakarta.annotation.PostConstruct
-import jakarta.annotation.PreDestroy
-import java.util.concurrent.ConcurrentHashMap
-import java.util.concurrent.atomic.AtomicLong
 import org.springframework.stereotype.Service
 import ua.lviv.iot.parkingServer.annotation.TrackMetrics
-import ua.lviv.iot.parkingServer.datastorage.ParkingFileStore
+import ua.lviv.iot.parkingServer.repository.ParkingMongoRepository
 import ua.lviv.iot.parkingServer.exception.EntityNotFoundException
 import ua.lviv.iot.parkingServer.model.Parking
 import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
 
 @TrackMetrics
 @Service
-class ParkingService(private val parkingFileStore: ParkingFileStore) : ParkingServiceInterface {
-    private val parking: MutableMap<Long, Parking> = ConcurrentHashMap()
-    private var id: AtomicLong = AtomicLong(1L)
+class ParkingService(private val parkingMongoRepository: ParkingMongoRepository) : ParkingServiceInterface {
+    override fun findAllEntities(): List<Parking> = parkingMongoRepository.findAll()
 
-    override fun findAllEntities(): List<Parking> = parking.values.toList()
+    override fun findEntityById(id: Long): Parking = parkingMongoRepository.findById(id)
+        .orElseThrow { EntityNotFoundException("Parking with id=$id not found") }
 
-    override fun findEntityById(id: Long): Parking =
-        parking[id] ?: throw EntityNotFoundException("Parking with id=$id not found")
-
-    override fun addEntity(entity: Parking): Parking {
-        val newId = id.getAndIncrement()
-        entity.id = newId
-        parking[newId] = entity
-        return entity
-    }
+    override fun addEntity(entity: Parking): Parking = parkingMongoRepository.save(entity)
 
     override fun updateEntity(id: Long, entity: Parking): Parking {
-        entity.id = id
-        parking[id] = entity
-        return entity
-    }
-
-    override fun deleteEntity(id: Long): Parking =
-        parking.remove(id) ?: throw EntityNotFoundException("Parking with id=$id doesn't exist")
-
-    @PostConstruct
-    override fun entityDataToHashMap() {
-        parkingFileStore.loadDataToHashmap(parkingFileStore, parking, id)
+        entity.id = findEntityById(id).id
```

Comment:
why trying to findEntityById and than get it id?

Reply:
My bad, fixed this workaround